java/vector/src/main/java/org/apache/arrow/vector/complex/ListVector.java

```diff
@@ -36,18 +39,24 @@
 import org.apache.arrow.vector.complex.impl.UnionListWriter;
 import org.apache.arrow.vector.complex.reader.FieldReader;
 import org.apache.arrow.vector.complex.writer.FieldWriter;
+import org.apache.arrow.vector.schema.ArrowFieldNode;
+import org.apache.arrow.vector.types.Types;
 import org.apache.arrow.vector.types.Types.MinorType;
 import org.apache.arrow.vector.types.pojo.Field;
 import org.apache.arrow.vector.util.CallBack;
 import org.apache.arrow.vector.util.JsonStringArrayList;
 import org.apache.arrow.vector.util.TransferPair;
 
+import com.google.common.collect.ImmutableList;
 import com.google.common.collect.ObjectArrays;
 
-public class ListVector extends BaseRepeatedValueVector {
+import io.netty.buffer.ArrowBuf;
+
+public class ListVector extends BaseRepeatedValueVector implements FieldVector {
 
-  UInt4Vector offsets;
+  final UInt4Vector offsets;// TODO: THis masks the same vector in the parent which is assigned to this in the constructor.
```

Comment:
I'll remove the TODO. That was more a personal note. I initially thought that the masked field had a different content but they actually have exactly the same content and I made it immutable so that there's no uncertainty whether it stays like that.

I'll open a separate bug for unsigned/signed of the offset vector.